sql/item_windowfunc.h

```diff
@@ -635,11 +636,12 @@ class Item_sum_ntile : public Item_sum_window_with_row_count
 
     longlong num_quantiles= get_num_quantiles();
 
-    if (num_quantiles <= 0) {
+    if (num_quantiles <= 0 || (num_quantiles != n_old_val_ && n_old_val_ != 0))
```

Comment:
Please change condition `n_old_val_ != 0` to `n_old_val > 0` or please elaborate when can be `n_old_val_ <0` ?

Reply:
n_old_val > 0 implemented